cache/memcached_client_test.go

```diff
@@ -158,6 +158,68 @@ func TestMemcachedClient_GetMulti(t *testing.T) {
 	})
 }
 
+func TestMemcachedClient_GetMultiWithError(t *testing.T) {
+	t.Run("returns error from backend", func(t *testing.T) {
+		backendErr := fmt.Errorf("connection failed")
+		backend := &erroringMockMemcachedClientBackend{
+			mockMemcachedClientBackend: newMockMemcachedClientBackend(),
+			err:                        backendErr,
+		}
+		client, err := newMemcachedClient(
+			log.NewNopLogger(),
+			backend,
+			&mockServerSelector{
+				servers: []mockServer{
+					{addr: "127.0.0.1:11211"},
+				},
+			},
+			MemcachedClientConfig{
+				Addresses:           []string{"localhost"},
+				MaxAsyncConcurrency: 1,
+				MaxAsyncBufferSize:  10,
+			},
+			prometheus.NewPedanticRegistry(),
+			"test",
+		)
+		require.NoError(t, err)
+
+		ctx := context.Background()
+		res, err := client.GetMultiWithError(ctx, []string{"foo"})
+		require.ErrorIs(t, err, backendErr)
+		require.Empty(t, res)
+	})
+
+	t.Run("returns partial results with error", func(t *testing.T) {
```

Comment:
This doesn't seem to be testing what it says it tests anymore since it looks like we're asserting that there's no error at all.